std/experimental/typecons.d

```diff
@@ -802,3 +802,223 @@ private template findCovariantFunction(alias finfo, Source, Fs...)
     else
         enum ptrdiff_t findCovariantFunction = x;
 }
+
+/++
+ + Type constructor for head-const variables.
+ +
+ + Head-const variables cannot be directly mutated or rebound, but references
+ + reached through the variable are typed with their original mutability.
+ + It is equivalent to $(D final) variables in D1 and Java, as well as
+ + $(D readonly) variables in C#.
+ +
+ + When $(D T) is a $(D const) or $(D immutable) type, $(D HeadConst) aliases
+ + to $(D T).
+ +/
+struct HeadConst(T)
+    if (!is(T == const) && !is(T == immutable))
+{
+    import std.typecons : Proxy;
+
+    private T HeadConst_value;
```

Comment:
I think this should begin lowercase like `headConst_value`, capitalization should be limited to static members or manifest constants IIRC.